src/syscheckd/syscheck_audit.c

```diff
@@ -652,10 +652,14 @@ void audit_parse(char *buffer) {
             // user_name & user_id
             if(regexec(&regexCompiled_uid, buffer, 2, match, 0) == 0) {
                 match_size = match[1].rm_eo - match[1].rm_so;
-                os_malloc(match_size + 1, w_evt->user_id);
-                snprintf (w_evt->user_id, match_size +1, "%.*s", match_size, buffer + match[1].rm_so);
-                const char *user = get_user("", atoi(w_evt->user_id), NULL);
-                w_evt->user_name = strdup(user);
+                char *uid = NULL;
+                os_malloc(match_size + 1, uid);
+                snprintf (uid, match_size +1, "%.*s", match_size, buffer + match[1].rm_so);
+                char *user = get_user("",atoi(uid), NULL);
+                os_strdup(user, w_evt->user_name);
```

Comment:
Please, assign the output of the `get_user` function to `w_evt->user_name`. It is not necessary to allocate and release memory (`user` variable).